src/lib/converter/plugins/PackagePlugin.ts

```diff
@@ -118,11 +123,23 @@ export class PackagePlugin extends ConverterComponent {
 
             if (result) {
                 this.readmeFile = result.file;
-                this.readmeContents = result.content;
+                this.readmeContents = this.processReadmeContents(
+                    result.content,
+                );
             }
         }
     }
 
+    private processReadmeContents(contents: string) {
+        if (this.stripYamlFrontmatter) {
+            return contents.replace(
+                /^[\r\n\s]*---\r?\n[\s\S]*?\r?\n---[\r\n\s]+/,
```

Comment:
Unless I'm missing something, isn't this equivalent to `/^\s*---\r?\n[\s\S]*?\r?\n---\s+/`?

This looks like it'll also catch the following, which I don't think it should?

```md
---
x
--- text here
```

Reply:
> Unless I'm missing something, isn't this equivalent to /^\s*---\r?\n[\s\S]*?\r?\n---\s+/?

Oh yeah, you're right. I'm not a specialist when it comes to regexes, I missed that whitespaces include newlines ;-)

> This looks like it'll also catch the following, which I don't think it should?

Good catch. GitHub does not render that as frontmatter. What do you think about this regex?

```
^\s*---\r?\n[\s\S]*?\r?\n---\s+\r?\n
```

Reply:
How about this?

```
^\s*---\r?\n[\s\S]*?\r?\n---\r?\n\s*
```

Having the newlines after required whitespace seems weird, I'm pretty sure we just care that `---` appears on a line by itself for closing frontmatter, then want to strip any leading whitespace.

Reply:
Hm, this does not match the frontmatter at all if there's a space after the last `---` (before the linebreak). Or do you think that this is out of spec?

Reply:
Ah, good catch, an extra `\s*?` there perhaps?

Reply:
Sorry for the delay. Why `*?`? I might be wrong, but I thought that `*` means `0 or more of the preceding token` making the `?` obsolete.

`^\s*---\r?\n[\s\S]*?\r?\n---\s+\r?\n`?
That is almost the one I posted in the first comment, just with a `*` instead of a `+` after the `---\s`.

Reply:
Yes, `*` means that, `*?` means do that same thing, but make the match non-greedy so that `\r?\n` matches earlier in the string. It shouldn't actually matter in this case

Reply:
Ah, makes sense. Didn't think about that. So `^\s*---\r?\n[\s\S]*?\r?\n---\s*?\r?\n\s*`?

Reply:
LGTM!

Reply:
Done